src/gridstack.ts

```diff
@@ -2251,8 +2251,8 @@ export class GridStack {
         if (wasAdded) {
           this.engine.cleanupNode(node); // removes all internal _xyz values
           node.grid = this;
+          delete node.grid._isTemp;
         }
-        delete node.grid._isTemp;

```

Comment:
To reproduce the issue:

1. Enable dragging a widget from outside onto the grid.
2. Ensure the grid is crowded, leaving very little space (less than the size of the widget).
3. Drag the widget into that small space.

This will cause the addition of the widget to fail and become unresponsive. If you need further assistance, feel free to reach out. Thank you!

Reply:
so this doesn't fix dropping into a full grid ? what does it fix then ?

Reply:
These changes should resolve the issue. The step provided is intended to demonstrate how to reproduce the issue before implementing these changes. I apologize if the explanation is unclear.

Reply:
The issue arises because 'node.grid' is populated only when the widget is added (line 2253). Therefore, if the widget fails to be added, an error occurs because the system attempts to delete 'node.grid._isTemp', which does not exist.

I'll aim to create a reproduction of the issue sometime next week, if that works for you.

Reply:
if that's the case delete node.grid?._isTemp; woudl have suffice. I'll need to debug this to see the implication of move it out...

Reply:
| The issue arises because 'node.grid' is populated only when the widget is added (line 2253)

no it could come from another grid as well... anyway the optional null check is safe and right thing to do, but I may need to exit sooner and not do some work, but for now that fixes the js error. thank you for the demo...